cacheMgr/cacheMgr.go

```
@@ -0,0 +1,110 @@
+package cacheMgr
+
+type CacheItem struct {
+	Value interface{}
+	readCh chan interface{}
+	writeCh chan interface{}
+}
+
+type CacheKeyValue struct {
+	Key int
+	Value interface{}
+}
+
+type CacheMap struct {
+	SharedMap map [int]CacheItem
+	readCh chan interface{}
+	readKeyCh chan int
+	writeCh chan CacheKeyValue
+	insertCh chan CacheKeyValue
+}
+
+type SafeItemAccess interface {
+	Reader() interface{}
+	Writer( interface{} )
```

Should return an error status

---

done